### PR TITLE
feat: migrated ng-app from .analog to .ag extension

### LINKED ISSUE #1371: Repo: Update Analog SFC app to use .ag extension instead of .analog extension by default
### Which scope/s are relevant/related to the feature request?

Don't know / other

### Information

We've added support for the `.ag` extension when using the Analog SFC. This would update the `ng-app` in the repo to use the new extension instead.

We're updating the Analog Language Tools and IDEA Plugin to support the extension also.

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by nickytonline**
## PR Checklist

Closes #1371

## What is the new behavior?

Ports all *.analog files in the ng-app project to the *.ag file extension. It also updates imports to use *.ag instead of *.analog now.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

I'm thinking the new file extension is a breaking change, but I leave that up to the core team to decide.

### CODE REVIEW

File: apps/ng-app/src/app/pages/about.page.ag [hunk 1]
```diff
@@ -1,13 +1,13 @@
 <script lang="ts">
-  import { RouteMeta } from '@analogjs/router';
-  import Hello from '../hello.analog' with { analog: 'imports' };
+import { RouteMeta } from '@analogjs/router';
```
Comment (nickytonline) [inline apps/ng-app/src/app/pages/about.page.ag:2]: I can undo these formatting changes in a bunch of these files. Not sure if that was the Analog VS Code extension that did that. 🤔

**Comment by nickytonline** [comment]
One question I had was for the vite-env.d.ts, do we want to remove the *.analog TS module?


```diff
/// <reference types="vite/client" />

...

- declare module '*.analog' {
-   const cmp = any;
-   export default cmp;
- }

declare module '*.ag' {
  const cmp = any;
  export default cmp;
}
```

**Comment by brandonroberts** [comment]
It's fine to remove it in the test app.

> One question I had was for the vite-env.d.ts in the ng-app project, do we want to remove the *.analog TS module since this is just a test app or would *.analog files potentially still be added that aren't SFCs?
> 
> 
> ```diff
> /// <reference types="vite/client" />
> 
> ...
> 
> - declare module '*.analog' {
> -   const cmp = any;
> -   export default cmp;
> - }
> 
> declare module '*.ag' {
>   const cmp = any;
>   export default cmp;
> }
> ```

**Review by brandonroberts — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks! @allcontributors add @nickytonline for code

**Pull request merged**